sklearn/preprocessing/_polynomial.py

```diff
@@ -1018,18 +1074,35 @@ def transform(self, X):
                         XBS_sparse = XBS_sparse.tolil()
                         XBS_sparse[:, :degree] += XBS_sparse[:, -degree:]
                         XBS_sparse = XBS_sparse[:, :-degree]
+
+                    # replace any indicated values with 0:
+                    extended_nan_indicator = np.repeat(
+                        nan_indicator[:, [i]], n_splines, axis=1
+                    )
+                    XBS_sparse[extended_nan_indicator] = 0
+
                 else:
                     XBS[:, (i * n_splines) : ((i + 1) * n_splines)] = spl(x)
+                    # replace any indicated values with 0:
+                    extended_nan_indicator = np.repeat(nan_indicator, n_splines, axis=1)
+                    XBS[extended_nan_indicator] = 0
```

Comment:
If `nan_indicator` is updated to work on a single column at time `_get_mask(x, np.nan)` as suggested above, then this line will need to be update to use the `i * n_splines) : ((i + 1) * n_splines)` slice again.